### ISSUE TITLE
Restoring kanboard from backup "breaks" all attached documents

### ISSUE THREAD

**Issue by nreojb** [fguillot/kanboard#1749]
_EDIT:
Changed title from "Restoring MySQl DB "breaks" all attached documents" to "Restoring kanboard from backup "breaks" all attached documents"_

---

Hello!

I've stumpled across a problem when I restore a MySQL DB. All attached documents are broken! You can see the attachements and click on "download" but when you try to open them they are either empty or you get a message that the file is broken.

This is what I did:

**System A - Windows**
* I export the whole MySQL DB using phpmyadmin
* I export the entire "data" folder of kanboard
* I export the conf.php

**System B - Linux/Debian**
* I installed a new version of kanboard
* In MySQL I created an empty DB "kanboard" and a user "kanboard" wich has all rights on the DB "kanboard" using the commands:
  * CREATE DATABASE kanboard CHARACTER SET utf8;
  * CREATE USER 'kanboard'@'localhost' identified by 'password';
  * GRANT ALL on kanboard.* to 'kanboard'@'localhost';
* I imported the whole MySQL DB using the command:
  * mysql -u root -p kanboard < /home/admin/kanboard.sql
* I replaced the new folder "data" with my exported folder "data"
* I put the exported config.php in place

After all of that everything is properly restored (users, setting, projects, tasks ...) BUT I can't open attachments because they're broken or they're empty.

I tried this at least 3 times and everytime the attached documents are broken.

Am I doing something wrong here? Or is this a bug?

Thank you in advance

**Comment by nreojb** [comment]
Hello again.

I REALLY worked the whole day an this and I'm not even close to an solution! But I think this is a importend thing to konw:

**I think that this is not a MySQL problem because it also happens with sqlite!**

**System A - Windows 7 / XAMMP**
* I installed a new version of kanboard
* I did not change or create a individual config.php therefore sqlite is going to be used
* I created a test project and some tasks
* I attached some different files to some tasks (doc, docx, xls, txt)
* Then I exported only the "data" folder!

**System B - Debian/Raspbian 8.0 (jessie) / nginx & php5-fpm**
* I installed a new version of kanboard
* I delet the folder "data" of the newly installed kanboard
* I put the exportet "data" folder into the kanboard folder

After all of that everything is properly restored (users, setting, projects, tasks ...) BUT I still can't open attachments because they're broken or they're empty.

So this is not a MySQL problem as I assumed (why I assumed that in the first place is beyond me :-) )

If I can provide you with additional information (log, zipfile of the data folder using sqlite ...) feel free to ask! Any help on that would be highly appreciated!

Thank you very much!

**Comment by nreojb** [comment]
I have some additional information.

I'm on the **original kanboard installation** (not the restored one):
Now when I delet an attached file (in the browser) the corresponding file in /kanboard/data/files/# is going to be deleted!

Now I'm on the maschine with the **restored kanboard installation**:
When I delet an attachment (also in the browser) the corresponding file in /kanboard/data/files/# is **NOT **going to be deleted!

So the problem is that the "**connection**" ist lost!

How can I fix this? And still: is this a bug or am I doing something wrong?

Thank you!

**Comment by seancarson** [comment]
In terms of your files not showing up try the following.  

cd /var/www/html  
sudo chown -R www-data:www-data kanboard/data

this give apache owner rights to the kanboard/data folder which holds your files.  

Reference the documentation for Ubuntu 14.04 http://kanboard.net/documentation/ubuntu-installation

**Comment by nreojb** [comment]
@seancarson 

Thanks for your answer.

Rights/permissions are not the problem in this case. I even tried chmod 777...
Owner/group of course is www-data for the folder "kanboard" and all subfolders.

Bye the way I'm not using apache but nginx.

**Comment by nreojb** [comment]
FINALLY!!!

I found the problem!

Since windows using backslashes "\" and unix-based systems using normal slashes "/" the problem is the path-decleration in the SQL database!

As I said I switched from a windows to a unix-based system. Therefore one has to correct the path-declecation in the SQL database.

Just in case someone has to deal with the same in the future I used this SQL command to substitute "\" with "/":

`UPDATE files SET path = REPLACE(path,"\","/");`

After this everything I working again!
